### PR TITLE
Crypto for est client

### PR DESCRIPTION
**Pull request by scnmzhaw**
added optional EST functions `GetCsrAttributes` and `GetCaCerts `with CLI commands `est csrattr` and `est cacerts`.

### CODE REVIEW

File: src/cli/cli_est_client.cpp [hunk 1]
```diff
@@ -362,6 +415,236 @@ void EstClient::CleanUpTemporaryBuffer(void)
     mPublicKeyTempLength  = sizeof(mPublicKeyTemp);
 }
 
+otError EstClient::PrintoutCsrAttributes(uint8_t *aData, const uint8_t *aDataEnd)
```
Comment (ChrIgiSta) [inline src/cli/cli_est_client.cpp]: what about move this function to the core module? (CsrAttributes to String)
save one mbed include in the cli module...

File: src/cli/cli_est_client.cpp [hunk 2]
```diff
@@ -36,12 +36,14 @@
 #if OPENTHREAD_ENABLE_EST_CLIENT
 
 #include <mbedtls/debug.h>
+#include <mbedtls/oid.h>
```
Comment (ChrIgiSta) [inline src/cli/cli_est_client.cpp]: Do we need mbedtls includes here?

File: src/core/est/est_client.cpp [hunk 3]
```diff
@@ -146,89 +144,111 @@ bool Client::IsConnected(void)
...
-    otError        error                   = OT_ERROR_NONE;
-    uint8_t        buffer[CSR_BUFFER_SIZE] = {0};
-    size_t         bufferLength            = CSR_BUFFER_SIZE;
-    uint8_t *      bufferPointer           = NULL;
-    Coap::Message *mCoapMessage            = NULL;
+    otError        mError                               = OT_ERROR_NONE;
+    uint8_t        mBuffer[EST_CERTIFICATE_BUFFER_SIZE] = {0};
+    size_t         mBufferLength                        = EST_CERTIFICATE_BUFFER_SIZE;
+    uint8_t *      mBufferPointer                       = NULL;
+    Coap::Message *mCoapMessage                         = NULL;
```
Comment (ChrIgiSta) [inline src/core/est/est_client.cpp]: see new conventions: https://github.com/openthread/openthread/blob/master/STYLE_GUIDE.md#naming
undo 'm' prefix for function local scope in all functions...

File: src/core/est/est_client.cpp [hunk 4]
```diff
@@ -440,6 +503,108 @@ void Client::SimpleEnrollResponseHandler(otMessage *aMessage, const otMessageInf
...
+void Client::GetCsrAttributesResponseHandler(otMessage *          aMessage,
+                                             const otMessageInfo *aMessageInfo,
+                                             otError              aResult)
+{
+    OT_UNUSED_VARIABLE(aMessageInfo);
+
+    otCoapCode       mCoapCode                                = otCoapMessageGetCode(aMessage);
+    otEstType        mType                                    = OT_EST_TYPE_NONE;
+    uint8_t          mMessage[EST_ATTRIBUTES_BUFFER_SIZE + 1] = {0};
+    uint32_t         mMessageLength                           = otMessageGetLength(aMessage) - otMessageGetOffset(aMessage);
```
Comment (ChrIgiSta) [inline src/core/est/est_client.cpp]: line length

File: src/cli/cli_est_client.cpp (hunk 1 above)
Comment (scnmzhaw) [reply src/cli/cli_est_client.cpp]: good idea.

File: src/core/est/est_client.cpp (hunk 3 above)
Comment (scnmzhaw) [reply src/core/est/est_client.cpp]: done

**Pull request merged**